extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php

```diff
@@ -100,29 +83,37 @@ private function blockStats($blockId) {
 
 		// parse block comment
 		if ($data['comment'] != '') {
-			$comment = ParserPool::parse($data['comment'], $this->wg->Title, new ParserOptions())->getText();
+			$comment = ParserPool::parse(
+				$data['comment'],
+				$this->getTitle(),
+				ParserOptions::newFromContext( $this->getContext() )
+			)->getText();
 		}
 		else {
 			$comment = '';
 		}
 
-		$table  = Xml::buildTable( array( $row ), $tableAttribs, $headers );
-		$table  = str_replace("</table>", "", $table);
-		$table .= "<tr><th>" . wfMsg('phalanx-stats-table-text') . "</th><td colspan='8'>" . htmlspecialchars( $data['text'] ) . "</td></tr>";
-		$table .= "<tr><th>" . wfMsg('phalanx-stats-table-reason')  ."</th><td colspan='8'>{$data['reason']}</td></tr>";
-		$table .= "<tr><th>" . wfMsg('phalanx-stats-table-comment')  ."</th><td colspan='8'>{$comment}</td></tr>";
-		$table .= "</table>";
-
-		$this->setVal('table', $table);
-		$this->setVal('editUrl', $this->phalanxTitle->getLocalUrl( array( 'id' => $data['id'] ) ));
+		$this->response->setValues( [
+			'firstRow' => $row,
+			'text' => $data['text'],
+			'reason' => $data['reason'],
+			'comment' => $comment,
+		] );
+		$this->setVal('editUrl', $this->phalanxTitle->getLocalURL( [ 'id' => $data['id'] ] ) );
+		$this->setVal( 'blockId', $blockId );
 
 		/* match statistics */
 		$pager = new PhalanxStatsPager( $blockId );
+		$pager->setContext( $this->getContext() );
 		$this->setVal('statsPager',
 			$pager->getNavigationBar() .
 			$pager->getBody() .
 			$pager->getNavigationBar()
 		);
+
+		// SUS-269: Add JS for unblock button
+		$out->addModules( 'ext.wikia.Phalanx' );
+		$out->addJsConfigVars( 'wgPhalanxToken', $this->getUser()->getEditToken() );
```

Comment:
Why not just use `mw.user.tokens.get('editToken');` instead of setting a new config var?